components/core/src/os/process/linux.rs

```diff
@@ -69,14 +68,16 @@ pub fn current_pid() -> u32 {
 }
 
 /// Determines if a process is running with the given process identifier.
-pub fn is_alive(pid: Pid) -> bool {
-    let process_group_id = unsafe { libc::getpgid(pid) };
-    process_group_id >= 0
+pub fn is_alive(pid: u32) -> bool {
+    match unsafe { libc::kill(pid as pid_t, 0) } {
+        0 => true,
+        _ => false,
+    }
 }
 
-pub fn signal(pid: Pid, signal: Signal) -> Result<()> {
+pub fn signal(pid: u32, signal: Signal) -> Result<()> {
```

Comment:
I looked at the standard library and saw how they preferred to speak about `PID`s and they prefer `u32` over a different type. I decided to follow suit with the standard library instead of this approach I had taken but I'm more than willing to swap it back!